plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerContainerNameGenerator.java

```diff
@@ -0,0 +1,128 @@
+
+import org.eclipse.che.api.core.model.machine.Machine;
+import org.eclipse.che.api.core.model.machine.MachineConfig;
+import org.eclipse.che.commons.annotation.Nullable;
+import org.eclipse.che.commons.user.User;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class is used for generation docker container name or parsing important information from docker container name.
+ *
+ * @author Alexander Andrienko
+ */
+public class DockerContainerNameGenerator {
+    private static final String WORKSPACE_ID_GROUP = "(?<workspaceId>workspace[0-9a-z]{16})";
+    private static final String MACHINE_ID_GROUP   = "(?<machineId>machine[0-9a-z]{16})";
+
+    private static final String  CONTAINER_NAME_REGEX   = "^" + WORKSPACE_ID_GROUP + "_" + MACHINE_ID_GROUP + "(_.*){2}$";
+    private static final Pattern CONTAINER_NAME_PATTERN = Pattern.compile(CONTAINER_NAME_REGEX);
+
+    /**
+     * Return generated name for docker container. Method generate name for docker container in format:
+     * <br><p>workspaceId + "_" + machineId + "_" + userName +"_" + machineName</p>
+     * Notice: if userName or machineName contains incorrect symbols for creation docker container, then we skip this symbols
+     *
+     * @param workspaceId
+     *         unique workspace id, see more (@link WorkspaceConfig#getId)
+     * @param machineId
+     *         unique machine id, see more {@link Machine#getId()}
+     * @param userName
+     *         name of the user who is docker container owner, see more {@link User#getName()}
+     * @param machineName
+     *         name of the workspace machine, see more {@link MachineConfig#getName()}
+     */
+    public String generateContainerName(String workspaceId, String machineId, String userName, String machineName) {
+        String containerName = workspaceId + '_' + machineId + '_' + userName + '_' + machineName;
+        return containerName.toLowerCase().replaceAll("[^a-z0-9_-]+", "");
+    }
+
+    /**
+     * Parse machine's {@code containerName} to get information about this container (like workspaceId, machineId).
+     * Notice: method doesn't parse information about userName or machineName,because we do not give guarantees
+     * about the integrity of this data(see more {@link #generateContainerName(String, String, String, String)})
+     *
+     * @param containerName
+     *         name of the container
+     * @return information about container
+     */
+    public ContainerNameInfo parse(@Nullable String containerName) {
+        containerName = containerName.replace("/", "");
+
+        Matcher matcher = CONTAINER_NAME_PATTERN.matcher(containerName);
+        if (matcher.matches()) {
+            String workspaceId = matcher.group("workspaceId");
+            String machineId = matcher.group("machineId");
+            return new ContainerNameInfo(workspaceId, machineId);
+        }
+        return null;
```

Comment:
I think that it is better to return Optional in that case than return null and mark method as those which returns nullable